chainer/training/triggers/once_trigger.py

```diff
@@ -0,0 +1,36 @@
+class OnceTrigger(object):
+
+    """Trigger based on the starting point of the iteration.
+
+    This trigger accepts only once at starting point of the iteration. There
+    are two ways to specify the starting point: only starting point in whole
+    iteration or recalled when training resumed.
+
+    Args:
+        recall_on_resume (bool): Whether the extension is recalled or not when
+            restored from a snapshot. It is set to ``False`` by default.
+    """
+
+    def __init__(self, recall_on_resume=False):
+        self._recall_on_resume = recall_on_resume
+        self._flag_called = False
+
+    def trigger(self, trainer):
+        if self._flag_called:
+            return False
+        self._flag_called = True
+        return True
+
+    @property
+    def skip_initialize(self):
```

Comment:
I understood this feature would be useful, but I feel it inconsistent with the existing triggers' behavior. Extensions with other triggers initialize even if they will not be triggered after resume. Do you think this feature is mandatory for `OnceTrigger`?

Reply:
Certainly, some other triggers can know easily that they will never triggered, such as `manual_schedule_trigger`.
And I wonder if there may be other uses of this information that they will never triggered.
How about adding property `never_triggered` to trigger objects, instead of `skip_initialize`?
- `trainer` observe `never_triggered` for initialize the extension or not
- Default value of `never_triggered` is `False` for almost triggers, such as iterative triggers.

Reply:
Ah, but now I think that the name `never_triggered` is confusing it means whether "the trigger will be never triggered" or "the trigger have never been triggered".
I will think better name for this property.

Reply:
Last commit adopts above property, `finished`.

Reply:
The property `finished` looks good. I would like to take this approach, but maybe it is safer to work on that in another PR. I want to keep this PR simply covering the new "once" trigger, and let the change on the base trigger API be separated into a dedicated issue/PR.

Reply:
OK, I agree with you.
Now I deleted `finished` property, and I will make another PR for this property.